### ISSUE TITLE
Focus indicator disappears after selecting nav link with keyboard

### ISSUE THREAD

**Issue by pownkel** [microsoft/fluentui#14089]
### Environment Information

- **Package version(s)**: 7.98.0
- **Browser and OS versions**:
   Chrome version 84.0.4147.89 (Official Build) (64-bit)
   Windows 10

### Describe the issue:

When navigating a Nav with keyboard while a screen reader (NVDA, Jaws, or narrator) is on, the focus indicator disappears after selecting a link and does not reappear again while using arrow keys to move to other links within the nav:

![left nav selector](https://user-images.githubusercontent.com/55459788/87813756-5b501780-c817-11ea-98d6-6507199a7bd9.gif)

### Please provide a reproduction of the issue in a codepen:

https://codepen.io/pownkel/pen/MWKPQRX

#### Actual behavior:

After focusing on one of the links and pressing the enter key, the focus indicator disappears. Afterwards, if focus is moved to other nav links with the arrow keys, no focus indicator appears. This occurs when using NVDA, Jaws, or narrator in either edge or chrome (for narrator it happens when in scan mode).

#### Expected behavior:

The focus indicator should still appear in the nav after selecting a link.

### Documentation describing expected behavior

https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_focus_discernable_predictable

**Comment by paulgildea** [comment]
@smhigley as FYI

**Comment by xugao** [comment]
@pownkel - you can observe the same issue [here](https://codepen.io/xugao/pen/xxZmLMq) using plain html. I'm afraid this is not a Component issue but is a screen reader issue.

**Comment by iamrafan** [comment]
@xugao Looks like the example you mentioned is a little different. The links in the code pen example are navigable using 'arrow' keys and can be activated using 'enter' key both with and without using screen readers. The focus indicator is not visible when using screen readers but the functionality remains the same.

**Comment by pownkel** [comment]
@xugao In the example you mentioned, if I change the href values to "#" so no page is opened, I can still see the focus indicator after selecting a link. We are using nav links with an onClick function instead of a target url, so no page is opened when the link is selected, hence why the focus indicator should still appear on the link after selecting it. This happens correctly in the plain html example, but not in the office fabric nav component.

**Comment by xugao** [comment]
@pownkel @iamrafan  - I still believe the issue is with NVDA. It aligned my [codepen](https://codepen.io/xugao/pen/xxZmLMq) to be more identical to our Nav (list of buttons). And you can see the focus border disappears. If you can spot anything we have done incorrectly in our nav, let me know.

**Comment by xugao** [comment]
closing as this is external. feel free to re-open if you need to follow-up
